lib/vicadmin/validate.go

```diff
@@ -209,14 +204,37 @@ func NewValidator(ctx context.Context, vch *config.VirtualContainerHostConfigSpe
 		v.DockerPort = fmt.Sprintf("%d", opts.DefaultTLSHTTPPort)
 	}
 
-	err = v.QueryDatastore(ctx, vch, sess)
+	err := v.QueryDatastore(ctx, vch, sess)
 	if err != nil {
 		log.Errorf("Had a problem querying the datastores: %s", err.Error())
 	}
 	v.QueryVCHStatus(vch, sess)
 	return v
 }
 
+// Obtain the VCH name from vsphere
+func VCHName(ctx context.Context, sess *session.Session) string {
+	defer trace.End(trace.Begin(""))
+
+	self, err := guest.GetSelf(ctx, sess)
+	if err != nil {
+		log.Errorf("Unable to get VCH name due to unknown self-reference: %s", err)
+		log.Infof("Setting the VCH name to %s", DefaultVCHName)
+		return DefaultVCHName
+	}
+
+	newVM := vm.NewVirtualMachineFromVM(ctx, sess, self)
+	vchName, err := newVM.Name(ctx)
+	if err != nil {
+		log.Errorf("Unable to get VCH name: %s", err)
+		log.Infof("Setting the VCH name to %s", DefaultVCHName)
+		return DefaultVCHName
+	}
+
+	log.Infof("Setting the VCH name to %s", vchName)
```

Comment:
Please either move this log message out of this function and rename the function to GetVCHName since it is a function and not a method (the Get is implied if it's a method because it would be called like v.VCHName() but as a bare function its functionality is unclear with the name given) or make this function into a _method_ `func (v *Validator) SetVCHName(ctx, sess) err {` and do `v.Hostname = vchName` here so that the log message is correct. Currently someone could call `VCHName()` to get the name of the VCH and it will log "Setting the VCH name" when nothing of the sort is actually happening. You have a few ways you can go with this, but it can't remain as it is.